zip-0311.rst

```diff
@@ -2,7 +2,436 @@
+  a single transaction. (Zero outputs is useful for z2t transactions.)
+- Payment disclosures can only be created by a sender of the transaction,
+  and are non-malleable (that is, no-one else can take one or more payment
+  disclosures and construct a new one that they would not have been able
+  to make independently).
+- A payment disclosure may be tied to a challenge, creating an interactive proof.
+- Senders with a Sapling address may choose to provably reveal that address as part of the
+  payment disclosure.
+- Senders are not required to remember any ephemeral data from the creation of a transaction
+  in order to produce a payment disclosure for that transaction. (Performance may be
+  improved if they do cache witness data for spent notes.)
+
+
+Conventions
+===========
+
+The following functions used in this ZIP are defined in the Zcash protocol specification:
+[#protocol]_
+
+- :math:`\mathsf{SpendAuthSig.RandomizePrivate}(α, \mathsf{sk})`,
+  :math:`\mathsf{SpendAuthSig.Sign}(\mathsf{sk}, m)`, and
+  :math:`\mathsf{SpendAuthSig.Verify}(\mathsf{vk}, m, σ)` [#protocol-concretespendauthsig]_
+
+We reproduce some notation and functions from [#protocol]_ here for convenience:
+
+- :math:`[k] P` means scalar multiplication of the elliptic curve point :math:`P` by the
+  scalar :math:`k`.
+
+- :math:`\mathsf{BLAKE2b}\text{-}\mathsf{256}(p, x)` refers to unkeyed BLAKE2b-256 in
+  sequential mode, with an output digest length of 32 bytes, 16-byte personalization
+  string :math:`p`, and input :math:`x`.
+
+We also define the following notation here:
+
+- :math:`[a..b]` means the sequence of values inclusive of :math:`a` and exclusive of
+  :math:`b`.
+
+- :math:`a\text{.len}()` means the length of the vector :math:`a`.
+
+
+Specification
+=============
+
+Payment disclosure data structure
+---------------------------------
+
+A payment disclosure has the following fields:
+
+- `txid`: Transaction id for the transaction `tx` being disclosed.
+
+- `msg`: A message field, which could contain a challenge value from the party to whom
+  the payment disclosure is directed.
+
+- :math:`\mathsf{saplingOutputs}`: A vector of the Sapling Outputs that we are disclosing
+  [0..tx.shieldedOutputs.len()]
+
+  - :math:`\mathsf{index}`: An index into `tx.shieldedOutputs`.
+  - :math:`\mathsf{ock}`: The outgoing cipher key that allows this output to be recovered.
+    [#protocol-saplingencrypt]_
+
+- :math:`\mathsf{saplingSpends}`: A vector of the Sapling Spends for which we are proving
+  spend authority [0..tx.shieldedSpends.len()]
+
+  - :math:`\mathsf{index}`: An index into `tx.shieldedSpends`.
+  - :math:`\mathsf{cv}`: A value commitment to the spent note.
+  - :math:`\mathsf{rk}`: A randomized public key linked to the spent note.
+  - :math:`\mathsf{zkproof_{spend}}`: A Sapling spend proof.
+  - [Optional] A payment address proof `addr_proof`:
+
+    - Any :math:`(\mathsf{d, pk_d})` such that :math:`\mathsf{pk_d} = [\mathsf{ivk}] GH(\mathsf{d})`
```

Comment:
GH was not defined above. The protocol spec calls it :math:`\mathsf{DiversifyHash}` (which includes the necessary personalisation and URS), defined in section 5.4.1.6.